### ISSUE TITLE
Aggregate query

### ISSUE THREAD

**Issue by saitheexplorer** [jeromelebel/MongoHub-Mac#115]
Are there plans to enable queries using the aggregation framework?

**Comment by jeromelebel** [comment]
Is it useful to keep map reduce if there is aggregation?

**Comment by MRigal** [comment]
If it is not a big deal, I would rather keep both. Just to be able to use old scripts written for Map/Reduce, and also to test some Map/Reduce behaviour produced by other software/frameworks. I think also that in a few cases, Map/Reduce is easier to use and therefore eventually faster if running time is anyway low.

**Comment by jeromelebel** [comment]
I've included aggregation in the beta version. I've done the minimum for the features. If you want to test it, first, you need to get the beta version (select it in the preference window). And then you will have to activate the aggregation in the preference window.

Let me know how it works so far.

**Comment by svenvarkel** [comment]
Hi!

Thanks for including this feature!

I'm having trouble specifying anything for the "output" field. It does not seem to accept any value.

I've tried o, 'o', {out:'o'} ... Nothing seems to work.

**Comment by nathanjordan** [comment]
excellent!

**Comment by MRigal** [comment]
Could you provide a link to the beta version ? I would be pleased to test it, but I prefer to keep the 3.0.x aside to work on production servers

**Comment by jeromelebel** [comment]
@MRigal:  You can also duplicate MongoHub on the disk, and upgrade one version and not the other one.

https://mongohub.s3.amazonaws.com/MongoHub-3.1b2.zip

@svenvarkel does it work better on the beta 2?

**Comment by henrytseng** [comment]
I'd also love to see this implemented.

**Comment by jeromelebel** [comment]
I've done it in the beta. But I have no feedback so far.

**Comment by svenvarkel** [comment]
Hi!

I've tried to use it but I don't know what to write to "Output:" field. It seems not to be taking any argument.

**Comment by jeromelebel** [comment]
You used Map/Reduce. You can activate the Aggregation tab in the preferences (advanced)

**Comment by svenvarkel** [comment]
Yes, you're right, it occurred in MapReduce. Aggregation view works fine in general. There could be some small things for better usability but in general it's fine:)

**Comment by jeromelebel** [comment]
Yes, I need at least to navigate in the cursor (with a next button). I would need also to add the popup button to automatically open documents (like in the find tab).
Do you have other things in mind?

**Comment by jeromelebel** [comment]
Should be fixed in 3.1. I let you verify this issue and close it.

**Comment by KylePDavis** [comment]
Seems to be working. Thanks!

**Comment by duozmo** [comment]
Aggregation works, but it doesn’t seem to have much in the way of error reporting. Syntax errors are caught, but runtime errors fail silently. E.g. if I try a `$group` operator [without an `_id` field](https://docs.mongodb.org/manual/reference/operator/aggregation/group/), `mongo` shell will duly admonish me, but MongoHub just says `1 documents` and presents a list of zero documents.

Separately, it would be handy if there were some indication in the UI that the `Pipeline` field is expecting to begin and end with square brackets `[]`.
